tests/scripts/thread-cert/mesh_cop.py

```diff
@@ -0,0 +1,832 @@
+#  CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF
+#  SUBSTITUTE GOODS OR SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS
+#  INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN
+#  CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE)
+#  ARISING IN ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE
+#  POSSIBILITY OF SUCH DAMAGE.
+#
+from binascii import hexlify
+import io
+import struct
+
+from enum import IntEnum
+
+class TlvType(IntEnum):
+    CHANNEL = 0
+    PAN_ID = 1
+    EXTENDED_PAN_ID = 2
+    NETWORK_NAME = 3
+    PSKC = 4
+    NETWORK_MASTER_KEY = 5
+    NETWORK_KEY_SEQUENCE_COUNTER = 6
+    NETWORK_MESH_LOCAL_PREFIX = 7
+    STEERING_DATA = 8
+    BORDER_AGENT_LOCATOR = 9
+    COMMISSIONER_ID = 10
+    COMMISSIONER_SESSION_ID = 11
+    SECURITY_POLICY = 12
+    GET = 13
+    ACTIVE_TIMESTAMP = 14
+    COMMISSIONER_UDP_PORT = 15
+    STATE = 16
+    JOINER_DTLS_ENCAPSULATION = 17
+    JOINER_UDP_PORT = 18
+    JOINER_IID = 19
+    JOINER_ROUTER_LOCATOR = 20
+    JOINER_ROUTER_KEK = 21
+    PROVISIONING_URL = 32
+    VENDOR_NAME = 33
+    VENDOR_MODEL = 34
+    VENDOR_SW_VERSION = 35
+    VENDOR_DATA = 36
+    VENDOR_STACK_VERSION = 37
+    UDP_ENCAPSULATION = 48
+    IPV6_ADDRESS = 49
+    PENDING_TIMESTAMP = 51
+    DELAY_TIMER = 52
+    CHANNEL_MASK = 53
+    COUNT = 54
+    PERIOD = 55
+    SCAN_DURATION = 56
+    ENERGY_LIST = 57
+    DISCOVERY_REQUEST = 128
+    DISCOVERY_RESPONSE = 129
+
+class MeshCopMessageType(IntEnum):
+    JOIN_FIN_REQ = 1,
+    JOIN_FIN_RSP = 2,
+    JOIN_ENT_NTF = 3,
+    JOIN_ENT_RSP = 4
+
+def create_mesh_cop_message_type_set():
+    return [ MeshCopMessageType.JOIN_FIN_REQ,
+             MeshCopMessageType.JOIN_FIN_RSP,
+             MeshCopMessageType.JOIN_ENT_NTF,
+             MeshCopMessageType.JOIN_ENT_RSP ]
+
+# Channel TLV (0)
+class Channel(object):
+
+    def __init__(self, channel_page, channel):
+        self._channel_page = channel_page
+        self._channel = channel
+
+    @property
+    def channel_page(self):
+        return self._channel_page
+
+    @property
+    def channel(self):
+        return self._channel
+
+    def __eq__(self, other):
+        common.expect_the_same_class(self, other)
+
+        return self._channel_page == other._channel_page and self._channel == other.__channel
+
+    def __repr__(self):
+        return 'Channel(channel_page={},channel={})'.format(self._channel_page, self._channel)
+
+    def to_hex(self):
+        return struct.pack('>BBBH', TlvType.CHANNEL, 3, self.channel_page, self.channel)
+
+
+class ChannelFactory(object):
+
+    def parse(self, data, message_info):
+        data_tp = struct.unpack('>BH', data.read(3))
+        channel_page = data_tp[0]
+        channel = data_tp[1]
+        return Channel(channel_page, channel)
+
+
+# PanId TLV (1)
+class PanId(object):
+    # TODO: Not implemented yet
```

Comment:
Better throw NotImplemented Error.

Reply:
Well, I think in most cases the tlv classes will not have a `__init__` function with no parameters. When one try to use an unimplemented tlv class he might look up the `__init__` function and he would find that's unimplemented.